samples/boards/stm32/backup_sram/src/main.c

```diff
@@ -24,14 +31,19 @@ int main(void)
 		return 0;
 	}
 
-	printk("Current value in backup SRAM (%p): %d\n", &backup_value, backup_value);
+	if (backup.magic != BACKUP_MAGIC) {
+		backup.magic = BACKUP_MAGIC;
+		backup.value = 0;
```

Comment:
It could be better to indicate when the value is being reset.
```suggestion
		backup.value = 0;
		printk("Invalid magic in backup SRAM structure - resetting value.\n");
```